07.ls_object/ls.rb

```diff
@@ -0,0 +1,90 @@
+#! /usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require 'optparse'
+require 'etc'
+require_relative 'ls_file'
+require_relative 'ls_directory'
+
+COLUMNS_NUMBER = 3
+
+def main
+  option = ARGV.getopts('arl')
+
+  display_files_in_directory('.', option)
+end
+
+def display_files_in_directory(directory_path, option)
+  directory = LsDirectory.new(directory_path, hidden_file_presence: option['a'], reversed_order: option['r'])
+  ls_files = directory.ls_files
+  return if ls_files.empty?
+
+  if option['l']
+    puts "total #{directory.blocks_sum}"
+    print_files_details(ls_files)
+  else
+    print_files(ls_files)
+  end
+end
+
+def print_files(ls_files)
+  rows_number = ls_files.length.ceildiv(COLUMNS_NUMBER)
+  aligned_files = align_array_size(rows_number, ls_files)
+  transposed_files = aligned_files.each_slice(rows_number).to_a.transpose
+
+  file_name_max_length = ls_files.map { |file| file.file_name.length }.max
+  transposed_files.each do |files|
+    files.each { |file| print "#{file&.file_name&.ljust(file_name_max_length)}\s\s" }
+    print "\n"
+  end
+end
+
+def align_array_size(rows_number, ls_files)
+  modulo_files = ls_files.length % rows_number
+  if modulo_files.zero?
+    ls_files
+  else
+    ls_files + Array.new(rows_number - modulo_files)
+  end
+end
+
+def print_files_details(ls_files)
+  stat_max_length = calc_max_length_of_file_stat(ls_files)
+  ls_files.each do |file|
+    puts build_file_detail(file, stat_max_length)
+  end
+end
+
+def calc_max_length_of_file_stat(files)
+  max = { nlink: 0, user: 0, group: 0, size: 0 }
+  files.each do |file|
+    max[:nlink] = file.link_count.to_s.length if max[:nlink] < file.link_count.to_s.length
+    max[:user] = file.owner_name.length if max[:user] < file.owner_name.length
+    max[:group] = file.group_name.length if max[:group] < file.group_name.length
+    max[:size] = file.byte_size.to_s.length if max[:size] < file.byte_size.to_s.length
+  end
+  max
+end
+
+def build_file_detail(file, stat_max_length)
+  "#{file.type}#{file.permission}#{display_extended_attribute(file)}\s" \
+    "#{file.link_count.to_s.rjust(stat_max_length[:nlink])}\s" \
+    "#{file.owner_name.ljust(stat_max_length[:user])}\s\s" \
+    "#{file.group_name.ljust(stat_max_length[:group])}\s\s" \
+    "#{file.byte_size.to_s.rjust(stat_max_length[:size])}\s" \
+    "#{build_time_stamp(file)}\s" \
+    "#{file.file_name}"
```

Comment:
```suggestion
    "#{file.name}"
```

「ファイルのファイルネーム」は冗長な感じがするのでnameだけでいいかも